pywbemtools/pywbemcli/_connection_repository.py

```diff
@@ -255,76 +355,83 @@ def _load_connections_file(self):
+                            ConnectionRepository.connections_group_name]
+                    except KeyError:
+                        raise ConnectionsFileLoadError(
+                            self._connections_file,
+                            'Missing YAML key name: "{0}"'.format(
+                                ConnectionRepository.connections_group_name))
+
+                    # Try getting the default_connection name
+                    try:
+                        self._default_connection_name = dict_[
+                            ConnectionRepository.default_connection_grp_name]
+                    except KeyError:
+                        gn = ConnectionRepository.default_connection_grp_name
+                        raise ConnectionsFileLoadError(
+                            self._connections_file,
+                            'Missing YAML key name: "{0}"'.format(gn))
+
+                    # Try to rebuild the PywbemcliServer object for each
+                    # server definition
                     try:
                         for name, svr in six.iteritems(connections_dict):
                             self._pywbemcli_servers[name] = \
                                 PywbemServer.create(
                                     replace_underscores=True, **svr)
                             self._loaded = True
                     except KeyError as ke:
-                        raise KeyError("Items missing from record {0} in "
-                                       "connections file. Exception {1}".format
-                                       (ke, self._connections_file))
+                        raise ConnectionsFileLoadError(
+                            self._connections_file,
+                            "Element: {} missing from svr name: {}".format(
+                                ke, name))
                     except TypeError as te:
-                        raise TypeError('Invalid object type in connections '
-                                        'file: "{0}"; server name: "{1}". '
-                                        'Item: {2}'.
-                                        format(self._connections_file, name,
-                                               te))
-                except ValueError as ve:
-                    raise ValueError("Invalid YAML in connections file {0}. "
-                                     "Exception {1}".format
-                                     (self._connections_file, ve))
+                        raise ConnectionsFileLoadError(
+                            self._connections_file,
+                            'Invalid object type server name: "{0}". '
+                            'Item: "{1}"'.format(name, te))
+
+                except (TypeError, yaml.YAMLError) as exc:
+                    raise ConnectionsFileLoadError(
+                        self._connections_file,
+                        'Invalid YAML. Exception: {0}'.format(exc))
```

Comment:
This exception handler is probably meant for `yaml.safe_load()`. I suggest to move it to be just around that call.